Add Azure Video Analyzer Resource (#12665)
hashicorp/terraform-provider-azurerm: website/docs/r/video_analyzer.html.markdown

  name                = "exampleidentity"
  resource_group_name = azurerm_resource_group.example.name
  location            = azurerm_resource_group.example.location
}

resource "azurerm_role_assignment" "contributor" {
  scope                = azurerm_storage_account.example.id
  role_definition_name = "Storage Blob Data Contributor"
  principal_id         = azurerm_user_assigned_identity.example.principal_id
}

resource "azurerm_role_assignment" "reader" {
  scope                = azurerm_storage_account.example.id
  role_definition_name = "Reader"
  principal_id         = azurerm_user_assigned_identity.example.principal_id
}

resource "azurerm_video_analyzer" "example" {
  name                = "exampleanalyzer"
  location            = azurerm_resource_group.example.location
  resource_group_name = azurerm_resource_group.example.name

  storage_account {
    id          = azurerm_storage_account.example.id
    identity_id = azurerm_user_assigned_identity.example.id
  }

  identity {
    type = "UserAssigned"
    identity_ids = [
      azurerm_user_assigned_identity.example.id
    ]
  }

  tags = {
    environment = "staging"
  }

  depends_on = [

>     azurerm_user_assigned_identity.example,

Reviewer: this should be implied from the two role assignments, which depend on this?

```suggestion
```

Author: Great catch

    azurerm_role_assignment.contributor,
    azurerm_role_assignment.reader,
  ]
}
```

## Argument Reference

The following arguments are supported:

* `name` - (Required) Specifies the name of the Video Analyzer. Changing this forces a new resource to be created.

* `resource_group_name` - (Required) The name of the resource group in which to create the Video Analyzer. Changing this forces a new resource to be created.

* `location` - (Required) Specifies the supported Azure location where the resource exists. Changing this forces a new resource to be created.

* `storage_account` - (Required) An `storage_account` blocks is defined below.

* `identity` - (Required) An `identity` block is defined below.

* `tags` - (Optional) A mapping of tags assigned to the resource.

---

A `storage_account` block supports the following:

* `id` - (Required) Specifies the ID of the Storage Account that will be associated with the Video Analyzer instance.

> * `identity_id` - (Required) ID for User Assigned Managed Identity resources to be assigned.

Reviewer: perhaps this'd be clearer as:

```suggestion
* `user_assigned_identity_id` - (Required) Specifies the User Assigned Identity ID which should be assigned to a access this Storage Account.
```

Author: This again, comes down to, do we want to make this change if other resources don't use this term?

Reviewer 2: As per the other thread, we may as well use `user_assigned_identity_id` as that will converge with other resources when we are able to update them

---

A `identity` block supports the following:

> * `type` - (Required) Specifies the type of Managed Service Identity that should be configured on this Video Analyzer. Possible value is `UserAssigned`. 

Reviewer: `Managed Service Identity` is being referred to as a `Managed Identity` in newer Azure API's (although the term's used interchangeably), perhaps it's worth doing the same here:

```suggestion
* `type` - (Required) Specifies the type of Managed Identity that should be configured on this Video Analyzer. At this time the only possible value is `UserAssigned`. 
```

> * `identity_ids` - (Required) A list of IDs for User Assigned Managed Identity resources to be assigned.

Reviewer: perhaps this'd read clearer as:

```suggestion
* `identity_ids` - (Required) A list of User Assigned Identity ID's to be assigned to this Video Analyzer.
```

---

## Attributes Reference

The following attributes are exported:

* `id` - The ID of the Video Analyzer.

## Timeouts

The `timeouts` block allows you to specify [timeouts](https://www.terraform.io/docs/configuration/resources.html#timeouts) for certain actions:

* `create` - (Defaults to 30 minutes) Used when creating the Video Analyzer.
* `update` - (Defaults to 30 minutes) Used when updating the Video Analyzert.
* `read` - (Defaults to 5 minutes) Used when retrieving the Video Analyzer.
* `delete` - (Defaults to 30 minutes) Used when deleting the Video Analyzer.

## Import

Video Analyzer can be imported using the `resource id`, e.g.

```shell

> terraform import azurerm_video_analyzer.analyzer /subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/mygroup1/providers/Microsoft.Media/videoanalyzers/analyzer1

Reviewer: (as above) ARM says segments should be camelCased so we should match that (since we're defining these ID's in Terraform)

```suggestion
terraform import azurerm_video_analyzer.analyzer /subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/mygroup1/providers/Microsoft.Media/videoAnalyzers/analyzer1
```
